index2.html

```diff
@@ -0,0 +1,1253 @@
+              check <var>url</var> against.
+            </li>
+          </ol>
+      </p>
+      <p><strong>Output</strong>: <code>true</code> or <code>false</code></p>
+      <p>
+        When filtering <a>NDEF message</a>s with identifiers associated with
+        <var>url</var>, based on <a>URL scope</a>, run the following steps
+        atomically.
+        <ol>
+          <li>
+            If <var>registeredScopes</var> is <code>undefined</code>, then let
+            <var>scope</var> be the calling <a>web app</a>'s
+            <a>document base URL</a>.
+          </li>
+          <li>Otherwise let <var>scope</var> iterate over values of
+            <var>registeredScope</var>.
+          </li>
+          <li>If <var>url</var> is a substring of <var>scope</var>,
+            return <code>true</code>.
+          </li>
+          <li>If the iteration is finished, return <code>false</code>.</li>
+        </ol>
+      </p>
+    </section> <!-- within-scope algorithm -->
+
+    <p>
+      When the <code><dfn id="widl-NfcAdapter-watch-Promise">
+      watch</dfn>()</code> method is invoked, the <a>user agent</a> MUST run
+      the following steps:
+      <ol id="steps-watch">
+        <li>Let <var>promise</var> be a new <a><code>Promise</code></a> object.
+        </li>
+        <li>
+          Return <var>promise</var> and continue the following steps
+          asynchronously.
+        </li>
+        <li>
+          If there are no sufficient permissions to use this method,
+          then reject <var>promise</var> with a new
+          <code><a>DOMException</a></code> object whose <var>name</var> property
+          is set to <code>"<a>SecurityError</a>"</code>,
+          and terminate these steps.
+        </li>
+        <li>
+          Parse the argument <var>options</var>. If it is invalid in the given
+          platform, then reject <var>promise</var> with a new
+          <code><a>DOMException</a></code> object whose <var>name</var> property
+          is set to <code>"<a>SyntaxError</a>"</code>, and terminate these steps.
+        </li>
+        <li>
+          Make a request to the underlying platform to listen to
+          <a>NDEF message</a>s, and set up <a>within-scope</a> filtering for the
+          given <a>URL scope</a>, and also set up filtering for accepted content
+          types, as described by the <var>options</var> parameter.
+          As a consequence, fire <code>read</code> events only for
+          messages <a>within-scope</a> and accepted content types.
+          If the request fails, then reject <var>promise</var> with a new
+          <code><a>DOMException</a></code> object whose <var>name</var> property
+          is set to <code>"NotSupportedError"</code> , and terminate these steps.
+        </li>
+        <li>
+          If the request is successful, then resolve <var>promise</var> with an
+          identifier for the listener, which later can be used with the
```

Comment:
now I wonder if the ES15 Symbol's should be used for identifier? need to research